File: src/main/prim/etc/_dy.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc

import org.nlogo.core.Syntax
import org.nlogo.api.Numbers
import org.nlogo.agent.Turtle
import org.nlogo.nvm.{ Context, Reporter }

class _dy extends Reporter {
  override def syntax =
    Syntax.reporterSyntax(
      ret = Syntax.NumberType,
      agentClassString = "-T--")
  override def report(context: Context): java.lang.Double =
    Double.box(report_1(context))
  def report_1(context: Context): Double = {
    val result = context.agent.asInstanceOf[Turtle].dy
    if (StrictMath.abs(result) < Numbers.Infinitesimal)
      0 else result
  }
}

File: netlogo-gui/src/main/prim/etc/_dy.java
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc;

import org.nlogo.agent.Turtle;
import org.nlogo.api.Constants;
import org.nlogo.api.Syntax;
import org.nlogo.nvm.Context;
import org.nlogo.nvm.Reporter;

public final strictfp class _dy extends Reporter {
  @Override
  public org.nlogo.core.Syntax syntax() {
    return Syntax.reporterSyntax
        (Syntax.NumberType(), "-T--");
  }

  @Override
  public Object report(Context context) {
    return report_1(context);
  }

  public double report_1(Context context) {
    double value = ((Turtle) context.agent).dy();
    return (StrictMath.abs(value) < Constants.Infinitesimal())
        ? 0 : value;
  }
}
